### DISCUSSION TITLE
Homogenization issue 3D

### DISCUSSION THREAD

**Discussion by eduardogrodriguez89** [idaholab/moose#26259] [Q&A Modules: Solid mechanics]
Hi, 
I want to determine the effective properties of two materials, a matrix with fiber. In #25408 I could determine it in 2D. 
Now I want to extract the effective stiffnesses in 3D. A cube with a sphere inside.

file:///home/erodriguez/Im%C3%A1genes/box.png

Using a coarse mesh the code works, but when I used a fine mesh the simulation got stuck on: 
```
 Currently Setting Up
  Initializing
    Initializing Equation Systems........................................................................
```
The found the same problem in #23483. I updated moose, but the problem still goes on.

[mesh_coarse.zip](https://github.com/idaholab/moose/files/13561971/mesh_coarse.zip)
[mesh_fine.zip](https://github.com/idaholab/moose/files/13561972/mesh_fine.zip)
[rve_action_example.zip](https://github.com/idaholab/moose/files/13561974/rve_action_example.zip)

Does this issue look familiar to anyone / any clues on what I am doing wrong here?

Thanks 
Eduardo

**Comment by GiudGiud** [comment]
Hello

The image did not upload well.
Is this a parallel run?
Is the hold up with a coarse mesh or the fine mesh?

Guillaume

**Reply by eduardogrodriguez89** [reply]
I run both, single-core and parallel. 

The hold up occurs with the fine mesh.

![box](https://github.com/idaholab/moose/assets/53190983/3d6eb7cf-6af7-4c9e-a92f-963f8aa0e301)

**Reply by maxnezdyur** [reply]
The stall you showed can happen if your periodic mesh sides don't match exactly. I bet if you turned off your periodic BCs, it would very quickly pass that part. If it does then you mostly likely have a small difference in the meshes that would need to be taken care of.

**Reply by eduardogrodriguez89** [reply]
Ok, thanks, I'll fix the mesh and I will try again.

**Reply by maxnezdyur** [reply]
I am assuming you ran the check I said about turning off the periodic BCs? I don't want to send you down a rabbit hole. If you still get stuck without periodic BCs then I think the error is elsewhere.

**Reply by eduardogrodriguez89** [reply]
I turn off the periodic BCs and it works.
